TLS migration script

srid opened the pull request:
WIP branch for the migration script.

/cc @voidlock

voidlock commented on src/tls_migration.erl:
```diff
@@ -0,0 +1,67 @@
+
+attempt_connection(#drain{id=DrainID, channel_id=ChannelID, uri=#ex_uri{authority=#ex_uri_authority{host=Host, port=Port}} = URI}) ->
+    io:format("Attempting connection to ~p:~p for drain ~p~n", [Host, Port, DrainID]),
+    case logplex_tlssyslog_drain:do_connect(Host, Port, URI, DrainID, ChannelID) of
```
We should test an #insecure connect first, to ensure the drain works in the first place. Followed by the newer set of options.

voidlock commented on src/tls_migration.erl:
```diff
@@ -0,0 +1,67 @@
+    ets:foldl(fun(Drain, _Acc) ->
+                      case should_migrate_drain(Drain) of
+                          true ->
+                              migrate_drain_maybe(Drain);
```
Would be nice to export `should_migrate_drain/1` and `migrate_drain_maybe/1` so they can be manually tested in a console session.

voidlock commented on src/tls_migration.erl:
```diff
@@ -0,0 +1,67 @@
+-include("logplex_drain.hrl").
+-include("logplex_logging.hrl").
+
+main(_) ->
```
Would be niced to pass in some options here like `dry_run` so the entire script can be run without saving any changes

voidlock commented on src/tls_migration.erl:
```diff
@@ -0,0 +1,67 @@
+                          _ ->
+                              ok
+                      end
+              end, notused, drains),
```
I would suggest passing in an empty list `[]` and using the accumulator. This way the failed drains can be returned to the caller and printed all at once instead of interleaved with everything else.

srid commented:
@voidlock In these recent commits I have addressed your comments, and more, except the connect delay for papertrail. Feel free to provide further feedback ... I'm still getting a hang of writing Erlang, and pretty sure my code is not yet as idiomatic it probably should be.
